Public/Src/Tools/SandboxedProcessExecutor/Executor.cs

```diff
@@ -0,0 +1,322 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.ContractsLight;
+using System.IO;
+using System.Threading.Tasks;
+using BuildXL.Native.IO;
+using BuildXL.Processes;
+using BuildXL.SandboxedProcessExecutor.Tracing;
+using BuildXL.Utilities;
+using BuildXL.Utilities.Instrumentation.Common;
+using BuildXL.Utilities.Tracing;
+
+namespace BuildXL.SandboxedProcessExecutor
+{
+    internal sealed class Executor
+    {
+        private const int ProcessRelauchCountMax = 5;
+
+        private readonly Configuration m_configuration;
+        private readonly LoggingContext m_loggingContext = new LoggingContext("BuildXL.SandboxedProcessExecutor");
+        public readonly TrackingEventListener TrackingEventListener = new TrackingEventListener(Events.Log);
+        private readonly Stopwatch m_telemetryStopwatch = new Stopwatch();
+        private OutputErrorObserver m_outputErrorObserver;
+
+        /// <summary>
+        /// Creates an instance of <see cref="Executor"/>.
+        /// </summary>
+        public Executor(Configuration configuration)
+        {
+            Contract.Requires(configuration != null);
+
+            m_configuration = configuration;
+    }
+
+        public int Run()
+        {
+            AppDomain.CurrentDomain.UnhandledException +=
+                (sender, eventArgs) =>
+                {
+                    HandleUnhandledFailure(eventArgs.ExceptionObject as Exception);
+                };
+
+            TelemetryStartup();
+
+            ExitCode exitCode = RunInternal();
+
+            TelemetryShutdown();
+
+            return (int)exitCode;
+        }
+
+        private void HandleUnhandledFailure(Exception exception)
+        {
+            // Show the exception to the user
+            ConsoleError(exception.ToString());
+            
+            // Log the exception to telemetry
+            if (AriaV2StaticState.IsEnabled)
+            {
+                Logger.Log.SandboxedProcessExecutorCatastrophicFailure(m_loggingContext, exception.ToString());

```

Comment:
>Logger.Log.SandboxedProcessExecutorCatastrophicFailure(m_loggingContext, exception.ToString()); [](start = 16, length = 95)

pull outside of if() to ensure you always log it to the console

Reply:
isn't that done on the first line in this file?

---
In reply to: [276484932](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276484932) [](ancestors = 276484932)

Reply:
Yes, like other tools, I want to start first with Console.WriteLine.

---
In reply to: [276822890](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276822890) [](ancestors = 276822890,276484932)